cl/persistence/beacon_indexes/interface.go

```diff
@@ -0,0 +1,22 @@
+package beacon_indexes
+
+import (
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/cl/cltypes"
+)
+
+type BeaconIndexer interface {
```

Comment:
these interfaces all need context as the first argument

also, they should probably have their return argument as something passed in, not something returned.